src/main/java/com/example/clone/controller/CommentController.java

```diff
@@ -0,0 +1,36 @@
+package com.example.clone.controller;
+
+import com.example.clone.model.Comment;
+import com.example.clone.repository.CommentRepository;
+import com.example.clone.repository.PostRepository;
+import com.example.clone.repository.UserInfoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import com.example.clone.service.CommentService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/comments")
+public class CommentController {
```

Comment:
Check this guide for designing REST API:

https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/

For this controller, I would define class level mapping '/comment' and remove mapping '/add' from the method because that's already defined by the used HTTP method

Reply:
I've also switched to noun-based endpoints and switched from params to path variables where its advised.